components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java

```diff
@@ -603,12 +607,34 @@ private AccessTokenDO createNewTokenBean(OAuthTokenReqMessageContext tokReqMsgCt
         newTokenBean.setValidityPeriodInMillis(tokReqMsgCtx.getValidityPeriod());
         newTokenBean.setValidityPeriod(tokReqMsgCtx.getValidityPeriod() / SECONDS_TO_MILISECONDS_FACTOR);
         newTokenBean.setTokenBinding(tokReqMsgCtx.getTokenBinding());
-        newTokenBean.setAccessTokenExtendedAttributes(tokenReq.getAccessTokenExtendedAttributes());
+        newTokenBean.setAccessTokenExtendedAttributes(
+                prepareAccessTokenExtendedAttributes(tokenReq.getAccessTokenExtendedAttributes(), tokReqMsgCtx));
         setRefreshTokenDetails(tokReqMsgCtx, existingTokenBean, newTokenBean, oauthTokenIssuer);
 
         return newTokenBean;
     }
 
+    private AccessTokenExtendedAttributes prepareAccessTokenExtendedAttributes(
+            AccessTokenExtendedAttributes accessTokenExtendedAttributes,
+            OAuthTokenReqMessageContext tokReqMsgCtx) {
+
+        boolean isUserSessionImpersonationEnabled = OAuthServerConfiguration.getInstance()
```

Comment:
make this method more generic

Reply:
Fixed with 79c6e716ce76831330afaae93c208a3bcb865c9a